cms/djangoapps/contentstore/utils.py

```diff
@@ -455,3 +457,16 @@ def get_visibility_partition_info(xblock):
         "has_selected_groups": has_selected_groups,
         "selected_verified_partition_id": selected_verified_partition_id,
     }
+
+
+def is_self_paced(course):
+    """
+    Returns True if course is self-paced, False otherwise.
+    """
+    if course:
+        try:
+            return course.self_paced and SelfPacedConfiguration.current().enabled
+        except AttributeError:
```

Comment:
Not Actually. `SelfPacedConfiguration` inherits `ConfigurationModel` which has `current()` method that creates a dict and returns it. So if it is not set, it returns `enabled=False` in the dict when we call `SelfPacedConfiguration.current()`